### DISCUSSION TITLE
Can only an ESP32 read the data from the inverter?

### DISCUSSION THREAD

**Discussion by DaradiciLevente** [wlcrs/huawei_solar#785] [Connecting to the inverter]
Could I extract the data from the inverter (directly from the sdongle, or directly from the wifi of the inverter) with an esp32 and with the routines from this Home Assistant integration? The idea is to be able to extract the data without the need for an HA server. Has anyone done something similar? I mention that I use HA and this integration works great! Just an ESP32 and the Huawei inverter.

**Comment by wlcrs** [comment]
https://github.com/jesusrop/esphome_huawei_sun2000

**Reply by DaradiciLevente** [reply]
Without wires connected between esp32 and inverter exclusively via wifi is it not possible?

**Comment by DaradiciLevente** [comment]
I mention that via Home Assistant, I managed to extract the data and display it on a display. https://youtu.be/iKTeAz46-Ds
But curiosity pushes me further. Is it possible exclusively via wifi without Home Assistant?
